### ISSUE TITLE
Bootstrap 5

### ISSUE THREAD

**Issue by mo8it** [flask-admin/flask-admin#2163]
Bootstrap 4 is not optimal in flask-admin. The Bootstrap 3 support is better. Maybe just drop Bootstrap 4 and start with Bootstrap 5 which has been released months ago? Thanks for the awesome package!

**Comment by zloyded** [comment]
+

**Comment by dagostinelli** [comment]
Not sure about the "not optimal" bit.  But an upgrade to BootStrap 5 would be appreciated.

**Comment by Mo8it** [comment]
> Not sure about the "not optimal" bit. But an upgrade to BootStrap 5 would be appreciated.

I did mean bugs like this which I did finally manage to fix: https://github.com/flask-admin/flask-admin/pull/2233

**Comment by ealtamir** [comment]
Anyone knows how hard would it be to override flask-admin templates and use Bootstrap 5?

**Comment by zloyded** [comment]
somebody?

**Comment by dagostinelli** [comment]
@ealtamir It doesn't look that hard.  Give it a whirl.

**Comment by Mo8it** [comment]
I would like to give a try, too. Or I could help @ealtamir . But I am afraid that it ends up like this tiny PR: https://github.com/flask-admin/flask-admin/pull/2233

A one liner is not merged yet, after 6 months...

**Comment by macnewbold** [comment]
Following also. We also would be interested in Bootstrap 5.

**Comment by samialfattani** [comment]
BootStrap 5 supports LTR (Left-To-Right) direction for RTL languages such as Arabic, Hebrew...etc. BootStrap 5 would be much appreciated.

**Comment by mariofix** [comment]
Any one want to collaborate on making this happen?
I've started playing with the bootswatch update to bootstrap 5.2.3

**Comment by Soham7777777** [comment]
This Flask-Admin project is so much useful, Can anyone please guide how to add support for bootstrap-5.3.3 it would be cool!

**Comment by mariofix** [comment]
It's not as easy as it sounds, I have about 60% advanced (I don't use inlines so I haven't got around that), but the main problem is that support for BS4 is not complete, so upgrading it to bs5 is a bit difficult.

**Comment by macnewbold** [comment]
Indeed. If there's a branch I could help test for you, or contribute to, I'd be happy to do so. I'm hoping we can get a new release out, since there's an important (to me) bugfix in master that isn't released yet, and this would be a great one to have released soon. It's been three years since Bootstrap 5 came out, and v4 is now six years old, and v3 is ten years old. It's time and I think we can get this working now.

**Comment by mariofix** [comment]
> Indeed. If there's a branch I could help test for you, or contribute to, I'd be happy to do so. I'm hoping we can get a new release out, since there's an important (to me) bugfix in master that isn't released yet, and this would be a great one to have released soon. It's been three years since Bootstrap 5 came out, and v4 is now six years old, and v3 is ten years old. It's time and I think we can get this working now.

#2350

**Comment by hasansezertasan** [comment]
> Anyone knows how hard would it be to override flask-admin templates and use Bootstrap 5?

I don't think it's too hard. Check out these projects:

- [tomasznajda/flask-adminlte](https://github.com/tomasznajda/flask-adminlte)
- [shijl0925/Flask-AdminLTE3](https://github.com/shijl0925/Flask-AdminLTE3)

**Comment by hasansezertasan** [comment]
> It's not as easy as it sounds, I have about 60% advanced (I don't use inlines so I haven't got around that), but the main problem is that support for BS4 is not complete, so upgrading it to bs5 is a bit difficult.

X-editable has some problems with BS5, so I used a fork.

Also, what do you mean by "support for BS4 is not complete"? It seems OK to me, what's missing?

**Comment by mariofix** [comment]
> > Anyone knows how hard would it be to override flask-admin templates and use Bootstrap 5?
> 
> I don't think it's too hard. Check out these projects:
> 
> * [tomasznajda/flask-adminlte](https://github.com/tomasznajda/flask-adminlte)

I didn't know about this one, i'll check it out.

> * [shijl0925/Flask-AdminLTE3](https://github.com/shijl0925/Flask-AdminLTE3)

I contribute to this project and because it was impossible to upgrade there, I decided to include it as a PR here :) i guess i'm back there

**Comment by hasansezertasan** [comment]
> I contribute to this project and because it was impossible to upgrade there, I decided to include it as a PR here :) i guess i'm back there

Why was it impossible to upgrade there? 🤓

**Comment by mariofix** [comment]
https://github.com/flask-admin/flask-admin/blob/14e24c970f0ee3a29add830612eee9a0b0ba5dcc/flask_admin/base.py#L495C13-L497C89
```python
            :param template_mode:
                Base template path. Defaults to `bootstrap2`. If you want to use
                Bootstrap 3 or 4 integration, change it to `bootstrap3` or `bootstrap4`.
```

At first i just tried to put None or bs5 but a folder with the name is needed.
This parameter loads the base javascript and styles to work with bootswatch, if I just ignore it and make the template regardless, i still get those javascript and styles.

This is where my journey began, tried to make it work (asked for comments here) and made the PR to merge the boostrap5 support.

**Comment by hasansezertasan** [comment]
> At first i just tried to put None or bs5 but a folder with the name is needed. This parameter loads the base javascript and styles to work with bootswatch, if I just ignore it and make the template regardless, i still get those javascript and styles, with the bugs and fixes for that specific version of bootstrap.

Hmm. I use `template_mode="boostrap4"` because it is irrelevant if you override the templates. On the other hand, you are supposed to add the Bootstrap5 Bootswatch files right?

> I'm not an expert in front-end, so if there is another workaround it and try to make it work with both javascript loaded (bootstrap 4 and 5), i'll take it, i guess eventually the team here will make their own upgrade.

Why not just override the `<script>` and `<link>` tags?
